### ISSUE TITLE
NavigationBar Search Function

### ISSUE THREAD

**Issue by Th1nkK1D** [wevisdemo/parliament-watch#34]
![image](https://github.com/wevisdemo/parliament-watch/assets/8860448/59566039-651d-4016-b2ae-d43edb7d57b2)

View [Figma file](https://www.figma.com/file/cdtZnPegHm5CGHWqIND5Im/Design_Phase01?type=design&node-id=1227-13731&mode=dev)

## Depended on
- #3 
- #12 
- #21 

## Starting point
- Take a look at `NavigationBar` component first. They implement the search input field but not the result dropdown.
- Add a new component `SearchResult` which is some child under the `NavigationBar`
- The component should take the search keyword props bind with the value from the search input.
- Just mock the data to be filtered with the keyword as a `const`. The data should have the following type: (only filter from the `name` property)
```typescript
interface SeachIndexes {
   politicians: {
     name: string;
     description: string;
   }[],
   bills: {
     name: string;
     status: BillStatus
   }[],
   votings: {
     name: string;
     result?: DefaultVotingResult
   }[]
}
```
- Bold the keyword in the result is nice to have. You can research if it's too difficult then it can be ignored for now.

**Comment by napatswift** [comment]
I want to work on this issue kub 🙋🏽‍♂️

**Comment by napatswift** [comment]
@Th1nkK1D I have a question kub. When searching documents (or items), does the search require an exact match of all the query terms, or does it match documents that contain some but not necessarily all of the query terms?

For example, if I search for "กก ขข", will it match a document that contains "กก" but not contains "ขข"?

And if I search for "นาง ปิยบุตร", will it match a document with "นายปิยบุตร นามสกุล" which contains the name "ปิยบุตร" but not "นาง"?

**Comment by Th1nkK1D** [comment]
> @Th1nkK1D I have a question kub. When searching documents (or items), does the search require an exact match of all the query terms, or does it match documents that contain some but not necessarily all of the query terms?
> 
> For example, if I search for "กก ขข", will it match a document that contains "กก" but not contains "ขข"?
> 
> And if I search for "นาง ปิยบุตร", will it match a document with "นายปิยบุตร นามสกุล" which contains the name "ปิยบุตร" but not "นาง"?

Will it be too complicated to put the exact match on the top, then if the exact match is less than the limit, we add the result that contains some keywords?

**Comment by napatswift** [comment]
It's not complicated but I think it will be more expensive than throwing out documents that don't contain certain query keywords (learned from [yourcandidates](https://election66.wevis.info/yourcandidates/)).

But I'll go with the approach that doesn't require all queries first. Thanks kub 🙏 .

**Comment by Th1nkK1D** [comment]
Merged #39
